DllImportGenerator/DllImportGenerator/Marshalling/SafeHandleMarshaller.cs

```diff
@@ -0,0 +1,230 @@
+                                                GenericName(Identifier("CreateSafeHandle"),
+                                                    TypeArgumentList(SingletonSeparatedList(info.ManagedType.AsTypeSyntax())))),
+                                            ArgumentList())));
+                    }
+                    break;
+                case StubCodeContext.Stage.Marshal:
+                    if (info.RefKind != RefKind.Out)
+                    {
+                        // <managedIdentifier>.DangerousAddRef(ref <addRefdIdentifier>);
+                        yield return ExpressionStatement(
+                            InvocationExpression(
+                                MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression,
+                                    IdentifierName(managedIdentifier),
+                                    IdentifierName(nameof(SafeHandle.DangerousAddRef))),
+                                ArgumentList(SingletonSeparatedList(
+                                    Argument(IdentifierName(addRefdIdentifier))
+                                        .WithRefKindKeyword(Token(SyntaxKind.RefKeyword))))));
+
+
+                        ExpressionSyntax assignHandleToNativeExpression = 
+                            AssignmentExpression(SyntaxKind.SimpleAssignmentExpression,
+                                IdentifierName(nativeIdentifier),
+                                InvocationExpression(
+                                    MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression,
+                                        IdentifierName(managedIdentifier),
+                                        IdentifierName(nameof(SafeHandle.DangerousGetHandle))),
+                                    ArgumentList()));
+                        if (info.IsByRef && info.RefKind != RefKind.In)
+                        {
+                            yield return ExpressionStatement(
+                                AssignmentExpression(SyntaxKind.SimpleAssignmentExpression,
+                                    IdentifierName(handleValueBackupIdentifier),
+                                    assignHandleToNativeExpression));
+                        }
+                        else
+                        {
+                            yield return ExpressionStatement(assignHandleToNativeExpression);
+                        }
+                    }
+                    break;
+                case StubCodeContext.Stage.GuaranteedUnmarshal:
+                    StatementSyntax unmarshalStatement = ExpressionStatement(
+                        InvocationExpression(
+                            MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression,
+                                ParseTypeName(TypeNames.System_Runtime_InteropServices_MarshalEx),
+                                IdentifierName("SetHandle")),
```

Comment:
Being a guest here, but happened to see this is the only that doesn't currently use nameof?

Reply:
The MarshalEx type isn't a type in the .NET Standard 2.0 contract, so we can't use `nameof` for any of its methods. It's currently in Ancillary.Interop. In the future its methods will move into dotnet/runtime in some fashion.